test/Microsoft.ML.Tests/Scenarios/Api/Estimators/PredictAndMetadata.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using Microsoft.ML.Data;
+using Microsoft.ML.RunTests;
+using Microsoft.ML.Trainers;
+using Xunit;
+
+namespace Microsoft.ML.Tests.Scenarios.Api
+{
+    public partial class ApiScenariosTests
+    {
+        /// <summary>
+        /// </summary>
+        [Fact]
+        void PredictAndMetadata()
+        {
+            var dataPath = GetDataPath(TestDatasets.irisData.trainFilename);
+            var ml = new MLContext();
+
+            var data = ml.Data.ReadFromTextFile<IrisData>(dataPath, separatorChar: ',');
+
+            var pipeline = ml.Transforms.Concatenate("Features", "SepalLength", "SepalWidth", "PetalLength", "PetalWidth")
+                .Append(ml.Transforms.Conversion.MapValueToKey("Label"), TransformerScope.TrainTest)
+                .Append(ml.MulticlassClassification.Trainers.StochasticDualCoordinateAscent(
+                    new SdcaMultiClassTrainer.Options { MaxIterations = 100, Shuffle = true, NumThreads = 1, }));
+
+            var model = pipeline.Fit(data).GetModelFor(TransformerScope.Scoring);
+            var engine = model.CreatePredictionEngine<IrisDataNoLabel, IrisPredictionNotCasted>(ml);
+
+            var testLoader = ml.Data.ReadFromTextFile(dataPath, TestDatasets.irisData.GetLoaderColumns(), hasHeader: true, separatorChar: ',');
+            var testData = ml.CreateEnumerable<IrisData>(testLoader, false);
+            // Slot names on top of Score column represent original labels for i-th value in Score array.
+            VBuffer<ReadOnlyMemory<char>> slotNames = default;
```

Comment:
I'd wish we could have a simpler way to access the info about "ranked list of predicted 'possible categories/lables' " without needing to use `VBuffer<ReadOnlyMemory<char>>` and other kind of "low level" types which are not familiar to .NET developers. For upcoming servions we should hide VBuffer so it looks a lot simpler and user friendly for developers.
​